In [7]:
import pandas as pd

# --- Assumptions for a Medium to Large-Scale BSF Farming Facility in Kenya ---

# Facility Scale: Approximately 10-20 tons of organic waste processed per day.
# Location: Rural area in Kenya, with good access to organic waste sources and labor.
# Yields: Assumed to be consistent with industry averages for optimized BSF farms.
# Price Points: Based on current market estimates for BSF products in East Africa, subject to market fluctuations.

print("Defining assumptions and cost/revenue structures...")

# 1. Estimated Start-up Costs (in USD)
startup_costs = {
    'Land Lease/Purchase (5 years)': 50000, # Assuming land lease over purchase for initial phase
    'Facility Construction (sheds, processing area)': 200000,
    'Equipment (trays, drying, grinding, harvesting, breeding units)': 150000,
    'Initial BSF Colony (eggs/larvae)': 10000,
    'Permits & Licenses': 5000,
    'Utilities Setup (water, electricity)': 15000,
    'Initial Working Capital (3 months op-ex)': 75000 # Placeholder, will be calculated more precisely later
}

# 2. Estimated Annual Operational Costs (in USD)
operational_costs = {
    'Feedstock (organic waste acquisition & transport)': 80000,
    'Labor (10-15 full-time staff)': 120000,
    'Utilities (electricity, water, internet)': 30000,
    'Maintenance & Repairs': 20000,
    'Consumables (trays, nets, PPE, cleaning supplies)': 15000,
    'Transportation & Logistics (product delivery)': 25000,
    'Marketing & Sales': 10000,
    'Insurance': 5000,
    'R&D/Quality Control': 10000
}

# 3. Estimated Annual Revenues (in USD)
# Assuming annual production volumes and market prices
revenue_streams = {
    'Dried BSF Larvae (protein meal)': {
        'production_volume_kg_pa': 50000,
        'market_price_usd_per_kg': 1.50
    },
    'BSF Frass (organic fertilizer)': {
        'production_volume_kg_pa': 150000,
        'market_price_usd_per_kg': 0.10
    },
    'BSF Oil': {
        'production_volume_kg_pa': 5000,
        'market_price_usd_per_kg': 3.00
    }
}

# Calculate totals
total_startup_cost = sum(startup_costs.values())
total_operational_cost = sum(operational_costs.values())

total_revenue = 0
for stream, details in revenue_streams.items():
    total_revenue += details['production_volume_kg_pa'] * details['market_price_usd_per_kg']

print("Calculations complete. Displaying results and reiterating assumptions...")

# Output results
print(f"\nTotal Estimated Start-up Cost: ${total_startup_cost:,.2f}")
print(f"Total Estimated Annual Operational Cost: ${total_operational_cost:,.2f}")
print(f"Total Estimated Annual Revenue: ${total_revenue:,.2f}")

print("\n--- Key Assumptions --- ")
print("Facility Scale: Approximately 10-20 tons of organic waste processed per day.")
print("Location: Rural area in Kenya, with good access to organic waste sources and labor.")
print("Yields: Assumed to be consistent with industry averages for optimized BSF farms.")
print("Price Points: Based on current market estimates for BSF products in East Africa, subject to market fluctuations.")
print("Cost Breakdown:")
for cost_type, costs in {'Start-up Costs': startup_costs, 'Annual Operational Costs': operational_costs}.items():
    print(f"  {cost_type}:")
    for item, value in costs.items():
        print(f"    - {item}: ${value:,.2f}")
print("Revenue Breakdown (Annual):")
for stream, details in revenue_streams.items():
    print(f"  - {stream}: {details['production_volume_kg_pa']} kg @ ${details['market_price_usd_per_kg']:.2f}/kg = ${details['production_volume_kg_pa'] * details['market_price_usd_per_kg']:,.2f}")


Defining assumptions and cost/revenue structures...
Calculations complete. Displaying results and reiterating assumptions...

Total Estimated Start-up Cost: $505,000.00
Total Estimated Annual Operational Cost: $315,000.00
Total Estimated Annual Revenue: $105,000.00

--- Key Assumptions --- 
Facility Scale: Approximately 10-20 tons of organic waste processed per day.
Location: Rural area in Kenya, with good access to organic waste sources and labor.
Yields: Assumed to be consistent with industry averages for optimized BSF farms.
Price Points: Based on current market estimates for BSF products in East Africa, subject to market fluctuations.
Cost Breakdown:
  Start-up Costs:
    - Land Lease/Purchase (5 years): $50,000.00
    - Facility Construction (sheds, processing area): $200,000.00
    - Equipment (trays, drying, grinding, harvesting, breeding units): $150,000.00
    - Initial BSF Colony (eggs/larvae): $10,000.00
    - Permits & Licenses: $5,000.00
    - Utilities Setup (water, elect

## Profitability Analysis

### Subtask:
Perform a profitability analysis by comparing the estimated costs and revenues. This will involve calculating potential profit margins, return on investment (ROI), or other relevant financial metrics to determine the financial viability of the venture.


**Reasoning**:
I will calculate the annual net profit, gross profit margin, and return on investment (ROI) using the previously computed total startup cost, annual operational cost, and total annual revenue. I will also handle potential division by zero for revenue in the gross profit margin calculation.



In [6]:
print("\n--- Profitability Analysis ---")

# 1. Calculate Annual Net Profit
annual_net_profit = total_revenue - total_operational_cost
print(f"Annual Net Profit: ${annual_net_profit:,.2f}")

# 2. Calculate Gross Profit Margin
if total_revenue > 0:
    gross_profit_margin = (annual_net_profit / total_revenue) * 100
    print(f"Gross Profit Margin: {gross_profit_margin:.2f}%")
else:
    gross_profit_margin = 0
    print("Gross Profit Margin: 0.00% (Cannot calculate, total revenue is zero or negative)")

# 3. Calculate Return on Investment (ROI)
if total_startup_cost > 0:
    roi = (annual_net_profit / total_startup_cost) * 100
    print(f"Return on Investment (ROI): {roi:.2f}%")
else:
    roi = 0
    print("Return on Investment (ROI): 0.00% (Cannot calculate, total startup cost is zero)")

# Store metrics for potential future use (optional)
profitability_metrics = {
    'annual_net_profit': annual_net_profit,
    'gross_profit_margin': gross_profit_margin,
    'roi': roi
}


--- Profitability Analysis ---
Annual Net Profit: $-210,000.00
Gross Profit Margin: -200.00%
Return on Investment (ROI): -41.58%


## Conclusion and Justification

### Subtask:
Based on the profitability analysis, conclude whether the Black Soldier Fly farming facility would be a profitable venture in Kenya, providing a detailed explanation and justification for the findings.


## Summary:

### Q&A
Based on the feasibility study, the Black Soldier Fly (BSF) farming facility in Kenya, under the stated assumptions, **would not be a profitable venture**.

### Data Analysis Key Findings
*   The **total estimated start-up cost** for a medium to large-scale BSF farming facility was estimated to be \$505,000.00. Key components included facility construction (\$200,000.00) and equipment (\$150,000.00).
*   The **total estimated annual operational cost** was calculated at \$315,000.00. Significant cost drivers included labor (\$120,000.00) and feedstock acquisition/transportation (\$80,000.00).
*   The **total estimated annual revenue** from the sale of dried BSF larvae, BSF frass, and BSF oil was \$105,000.00. Dried BSF larvae contributed \$75,000.00, BSF frass \$15,000.00, and BSF oil \$15,000.00.
*   The profitability analysis revealed a **negative annual net profit of -\$210,000.00**.
*   The **gross profit margin** was calculated at **-200.00%**, indicating that operational costs significantly outweigh revenues.
*   The **Return on Investment (ROI)** was **-41.58%**, signaling that the venture would erode capital rather than generate returns under the current estimates.

### Insights or Next Steps
*   **Re-evaluate and Optimize Cost Structures & Revenue Potential**: Given the significant unprofitability, it is crucial to thoroughly re-examine all cost assumptions for potential reductions (e.g., lower-cost feedstock sourcing, energy efficiency, labor optimization) and simultaneously investigate higher-value revenue streams or increased production volumes, possibly exploring premium markets or additional products like chitin.
*   **Conduct In-depth Market Research and Scale Optimization**: Undertake more detailed market research to validate product pricing and demand for BSF products in Kenya and East Africa, and perform a techno-economic assessment to identify an optimal facility scale and technology that could achieve better cost efficiency and profitability. This should also include exploring potential government incentives or subsidies for sustainable agriculture.
